docs/js/custom-active-focus-traps.js

```diff
@@ -0,0 +1,25 @@
+const { createFocusTrap } = require('../../index');
+
+module.exports = () => {
+  const container = document.getElementById('custom-active-focus-traps');
+
+  const activeFocusTraps = {
+    activateTrap: (trap) => {
+      container.classList.add('is-active');
+    },
+    deactivateTrap: (trap) => {
+      container.classList.remove('is-active');
+    },
+  };
```

Comment:
@skjnldsv In the override here, you're basically doing nothing, which means you're _preventing_ the trap that's already active from being paused when another one is being activated (or unpaused), and you're _preventing_ a previously-activate-and-now-paused trap from being unpaused when a trap gets deactivated.

Is this your real-world scenario, or just for the sake of the test?

If it's your real-world scenario (i.e. what you're actually doing in your override), how are you still not ending up with traps competing for focus? You still have 2 traps that have each added their own listeners to the document (via their own instances of `addListeners()`) and are both listening for focus events and wanting to pull focus into one another, away from the other one.

Reply:
This must just be for the test, and it's not demonstrating anything meaningful in terms of why the feature exists. You should at least be creating 2 traps in here and doing something a bit more meaningful.

I'd still like to know what you're doing in your override that's actually preventing the two instances of focus-trap from fighting for focus so I can get a clearer mental picture.

I don't want to expose the guts of the whole internal state if possible. I'd like to find a better, more flexible solution that would be easier for me to maintain into the future.

Reply:
Think about 2 isolated bundles, each one with this piece of code:

```ts
import { createFocusTrap } from 'focus-trap'

// --
```

We will have `app-a.js` and `app-b.js` 

Now think about @skjnldsv example.

Each bundle have an isolated instance of the function `createFocusTrap` and inside this bundle an isolate reference of `activeFocusTraps`

The object `activeFocusTraps` is a `closure` with a private member/state `trapQueue`
Now we have 2 `trapQueue` isolated from each other.

When `app-a.js` create an instance of `focus-trap` and active it, the `trapQueue` will receive it.
If `app-a.js` create another instance of `focus-trap`, everything will work.

But, if `app-b.js` creates an instance of `focus-trap` its state (`trapQueue`) do not know about `app-a.js` instance, so when `app-b.js` activate its instance while the instance of `app-a.js` is enabled, we will face an infinite loop.

That is exactly what is happening in our ecosystem.

Reply:
@vinicius73 Yes, thank you, I understand all that. I've already worked out how you end-up in the conflict.

I think I can guess what you're doing with this new option, and that's to override both focus-trap instances and point both to the same implementation where you replicate the queue yourselves.

I will think about this some more.